dashboard/app/controllers/api/v1/teacher_feedbacks_controller.rb

```diff
@@ -0,0 +1,44 @@
+class Api::V1::TeacherFeedbacksController < Api::V1::JsonApiController
+  load_and_authorize_resource
+
+  # Use student_id, level_id, and teacher_id to lookup the feedback for a student on a particular level and provide the
+  # most recent feedback left by that teacher
+  def show_feedback_from_teacher
+    @feedback = TeacherFeedback.where(student_id: params[:student_id], level_id: params[:level_id], teacher_id: params[:teacher_id]).last
+    render json: @feedback
+  end
+
+  # Use student_id and level_id to lookup the most recent feedback from each teacher who has provided feedback to that
+  # student on that level
+  def show_feedback_for_level
+    query = <<-EOS
+      SELECT b.*
+      FROM (
+        SELECT teacher_id, max(created_at) as created_at
+        FROM teacher_feedbacks GROUP BY teacher_id) a
+      INNER JOIN teacher_feedbacks b ON a.teacher_id = b.teacher_id AND a.created_at = b.created_at
+      WHERE b.student_id = #{params[:student_id]} and b.level_id = #{params[:level_id]}
+    EOS
```

Comment:
Why all this explicit SQL?

First of all, a complex and common query should be defined as a scope or class method on the model with unit tests. Generally any sufficiently complex logic should go in the model and be tested. We're striving for skinny controllers, fat models. 

Secondly, why not leverage ActiveRecord here?
And why is this query joining TeacherFeedbacks with itself? 

AFAICT we just want to retrieve the most recent (latest `created_at`) `TeacherFeedback` for the given `student_id` and `level_id`. If so, that can be done like this:
```ruby
@level_feedback = TeacherFeedback.where(
  student_id: params[:student_id], 
  level_id: params[:level_id]
).order('created_at DESC').first
```

Reply:
These are all great questions - I spent a few days trying to figure out how to do this in ActiveRecord and eventually just used SQL to get myself un-blocked from this part. The goal of this method is to find the most recent comment from each teacher that has this student in their section. The query joins TeacherFeedbacks with itself to first find the most recent 'created_at' date for each teacher and then to find the associated comment with that teacher/timestamp combo. I'll see what I can do to make this a skinny controller.